File: solutions/python/allergies/1/allergies.py
class Allergies:
    ingredients = {
        "eggs": 1,
        "peanuts": 2,
        "shellfish": 4,
        "strawberries": 8,
        "tomatoes": 16,
        "chocolate": 32,
        "pollen": 64,
        "cats": 128,
    }

    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        return bool(self.score & Allergies.ingredients.get(item))

    @property
    def lst(self):
        return [
            ingredient
            for ingredient in Allergies.ingredients
            if self.allergic_to(ingredient)
        ]

File: solutions/cpp/allergies/1/allergies.cpp
#include "allergies.h"

namespace allergies {
    allergy_test::allergy_test(int score) {
        const std::unordered_map<int, std::string> allergens({
            {1, "eggs"},
            {2, "peanuts"},
            {4, "shellfish"},
            {8, "strawberries"},
            {16, "tomatoes"},
            {32, "chocolate"},
            {64, "pollen"},
            {128, "cats"},
        });
        for (auto x : allergens) {
            if ((score & x.first) == x.first)
                list_of_items_allergic_to.insert(x.second);
        }
    }
    std::unordered_set<std::string> allergy_test::get_allergies() {return list_of_items_allergic_to;}
    bool allergy_test::is_allergic_to(std::string str) {
        return list_of_items_allergic_to.count(str);
    }
}  // namespace allergies
